docker-compose.yml

```diff
@@ -0,0 +1,10 @@
+version: "3.9"
+
+services:
+  web:
+    image: dataengbook
+    build:
+      context: .
+      dockerfile: Dockerfile
+    ports:
+      - "80:80"
```

Comment:
I think we could add volume mount here so the docker webapp can do live reload

Reply:
I'm not sure I used to do something like
```
services:
  web:
    image: dataengbook
    build:
      context: .
      dockerfile: Dockerfile
    ports:
      - "80:80"
    volumes:
      - .:/usr/src/app
```
but the web app still does not reload
maybe I have to write another Dockerfile for live reload for dev environment

but I have no idea how to do it. Could you suggest me?

I use docker because my pc does not have npm or yarn.

Reply:
that's because in your Dockerfile, it only serves static resources
https://github.com/thedataengineeringbook/thedataengineeringbook/blob/086d87bccffd13921a7062961cf23854f30e26aa/Dockerfile#L14-L17

for live reload to work, you need to run it as a node server

try this:
```dockerfile
# Dockerfile

FROM node:18

WORKDIR /app

COPY package*.json /app
COPY yarn.lock /app

RUN yarn install

COPY . /app

CMD [ "yarn", "start" ]
```


```yaml
# docker-compose.yml

version: "3.9"

services:
  dataengbook:
    container_name: dataengbook
    build:
      context: .
      dockerfile: Dockerfile
    ports:
      - "3000:3000"
    volumes:
      - ./docs:/app/docs
```

Also in `package.json`, need to set start directive as `    "start": "docusaurus start --host 0.0.0.0",`

Reply:
Great, it's work! Thank you a lot. But I have 2 questions.

1. Should I set `"start": "docusaurus start --host 0.0.0.0"` in `package.json`?  
Or someone who like to test in docker env must change it locally

2. It will reload after we restart container right? 
`docker restart dataengbook`

Reply:
> Should I set `"start": "docusaurus start --host 0.0.0.0"` in `package.json`?

By default, docusaurus only allow access only via the same instance as the server. IE you can't access docusaurus server running on another server. By setting `--host 0.0.0.0` this means it allows access to the docusaurus from any terminal

> It will reload after we restart container right
No need. If you only change markdown files, the changes would be updated automatically without performing docker restart. However, if you update `sidebar.js`, you still have to rebuild the docker image and start a new container instance. Or you could add another volume mapping for `sidebar.js` so you would only need to restart the container

Reply:
Thank you very much, I already added docker-compose.dev.yml from your suggestion